pkg/controller/machine_test.go

```diff
@@ -322,9 +312,48 @@ func TestController_defaultContainerRuntime(t *testing.T) {
 			}
 
 			cr := machine.Spec.Versions.ContainerRuntime
+
 			if diff := deep.Equal(cr, test.resCR); diff != nil {
 				t.Errorf("expected to get %s+%s instead got: %s+%s", test.resCR.Name, test.resCR.Version, cr.Name, cr.Version)
 			}
 		})
 	}
 }
+
+func createTestMachineController(t *testing.T, objects ...runtime.Object) (*Controller, *machinefake.Clientset) {
+	kubeClient := kubefake.NewSimpleClientset()
+	machineClient := machinefake.NewSimpleClientset(objects...)
+
+	kubeInformersFactory := kubeinformers.NewSharedInformerFactory(kubeClient, time.Millisecond*50)
+	kubeSystemInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, time.Millisecond*50)
+	machineInformerFactory := machineinformers.NewSharedInformerFactory(machineClient, time.Millisecond*50)
+
+	machineIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+	for _, o := range objects {
+		if err := machineIndexer.Add(o); err != nil {
```

Comment:
this should not be needed as the listers get filled with the objects from the client automatically when starting the informers